src/main/java/com/artipie/asto/memory/InMemoryStorage.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.memory;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.common.primitives.Bytes;
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Flow;
+import org.reactivestreams.FlowAdapters;
+
+/**
+ * Simple implementation of Storage that holds all data in memory.
+ *
+ * @since 0.1
+ */
+public final class InMemoryStorage implements Storage {
+
+    /**
+     * Values stored by key strings.
+     */
+    private final Map<String, byte[]> data = new HashMap<>();
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
```

Comment:
@olegmoz seems that `exists` is pretty easy to implement. Why don't we do it?

Reply:
@iakunin I was trying to not make this PR too big (it's 200 lines as it is). It's simple, but requires test coverage etc

Reply:
@olegmoz you've done it absolutely right, thanks :) Let's add the pdd-puzzle just not to forget implement the other methods.

Reply:
@iakunin That is going to be part of issue #59 , do not want to create more issues

Reply:
@olegmoz but this PR is already solves https://github.com/artipie/asto/issues/59. I thought that we are guided by the principle "one issue - one pull-request" according to [PDD methodology](https://www.yegor256.com/2010/03/04/pdd.html).

In my opinion it's better to leave a puzzle(s) for the remaining parts of the issue https://github.com/artipie/asto/issues/59 and close the https://github.com/artipie/asto/issues/59. After that @0pdd will create issues on each puzzle you left so you can start developing the next part of the original issue.

Reply:
@iakunin I am not sure we are following this principle in Artipie. @g4s8 what do you think about it? Should I create more issues and close this one?

Reply:
@olegmoz you can either add a todo markers to submit next tickets automatically and close the base ticket, or you can submit multiple PRs for one ticket to solve it without closing

Reply:
@g4s8 @iakunin so I'd prefer the second option, to have multiple PRs for single issue. May we proceed with this PR?

Reply:
@olegmoz yes, you can submit multiple PRs for one issue.

Reply:
@olegmoz no problem: just leave the original issue opened then.